### PR TITLE
refactor(daemon): typed startup phases and HostLayout dedup (T1, T4)

### PR DESCRIPTION
**Pull request by AprilNEA**
## Summary

ABX-263 (T1: startup contract) + ABX-266 (T4: host layout dedup).

- Extract `HostLayout` into `arcbox-constants` as single source of truth for host paths — eliminates 3 duplicate `resolve_data_dir()` implementations across daemon, CLI daemon command, and CLI logs command
- Remove `DaemonContext::runtime()` panic path — post-runtime callers receive `&Arc<Runtime>` as explicit parameter; `init_runtime()` returns `Arc<Runtime>` directly
- Split `DaemonContext` into `EarlyContext` (pre-lock) → `DaemonContext` (lock held) — `daemon_lock: DaemonLock` is non-optional, making it a compile error to skip lock acquisition
- CLI `daemon.rs` now uses `HostLayout` for all path resolution instead of manual construction

## Test plan

- [x] `cargo test -p arcbox-cli -p arcbox-daemon -p arcbox-constants`
- [x] `cargo clippy -p arcbox-cli -p arcbox-daemon -p arcbox-constants --all-targets -- -D warnings`
- [x] `cargo fmt --check`

### CODE REVIEW

File: common/arcbox-constants/Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,12 @@ repository.workspace = true
 authors.workspace = true
 description = "Shared protocol and runtime constants for ArcBox"
 
+[features]
+default = []
```
[the remark this answers is not in the document]
Comment (AprilNEA) [reply common/arcbox-constants/Cargo.toml:13]: Not needed — workspace `Cargo.toml` already specifies `features = ["std"]`, so `cargo test --workspace` (which is what CI runs) picks it up. Running bare `cargo test -p arcbox-constants` without features is an edge case not worth optimizing for.

File: Cargo.toml [hunk 2]
```diff
@@ -175,7 +175,7 @@ macos-resolver = "0.2.0"
 fc-sdk = "0.2.3"
 
 # Internal crates
-arcbox-constants = { version = "0.3.15", path = "common/arcbox-constants" } # x-release-please-version
+arcbox-constants = { version = "0.3.15", path = "common/arcbox-constants", features = ["std"] } # x-release-please-version
```
[the remark this answers is not in the document]
Comment (AprilNEA) [reply Cargo.toml:178]: Checked — `guest/arcbox-agent` and `rpc/arcbox-transport` inherit `["std"]` via `workspace = true`, but `dirs` is pure Rust path lookup with no platform issues for musl cross-compilation. Neither crate uses `HostLayout`; they only consume plain constants. The overhead is negligible and not worth splitting the feature configuration for.

File: app/arcbox-daemon/src/startup/mod.rs [hunk 3]
```diff
@@ -65,21 +66,30 @@ pub async fn init_early(args: DaemonArgs) -> Result<DaemonContext> {
...
 /// Fast when no stale daemon exists. If a stale daemon holds the lock,
 /// sends SIGTERM (with 30 s grace period and SIGKILL fallback), then
 /// blocks on `flock(LOCK_EX)` until the lock is released. Must run
 /// before `start_grpc` because the old daemon may still be listening
 /// on the same socket paths.
-pub async fn acquire_lock(ctx: &mut DaemonContext) -> Result<()> {
-    let run_dir = ctx.data_dir.join(arcbox_constants::paths::host::RUN);
+pub async fn acquire_lock(early: EarlyContext) -> Result<DaemonContext> {
+    let run_dir = early.layout.run_dir.clone();
     let lock = tokio::task::spawn_blocking(move || DaemonLock::acquire(&run_dir))
```
[the remark this answers is not in the document]
Comment (AprilNEA) [reply app/arcbox-daemon/src/startup/mod.rs]: Fixed — `DaemonLock::acquire()` now takes the full `lock_path: &Path` instead of `run_dir`. The caller passes `early.layout.lock_file` directly.
